algorithm/algorithm.md

```diff
@@ -0,0 +1,147 @@
+# MPM Algorithm
+
+## Main Algorithm
+
+The Material Point Method (MPM) algorithm comprises of 4 major parts.
+
+> **Note** 
+> * MPM stores information at the material points and not at the nodes as it is in Finite Element Method. This includes the application of traction at the particles, not at the nodes.
+> * The code is implementing Modified Update Stress Last (MUSF). The other option is to do Update Stress First (USF) or to do traditional Update Stress last. It is done by moving this block of code to the bottom after the material points velocities and displacements have been updated.
+> * The code computes material points mass in the initialization. Even though the density is updated later, the mass is conserved.
+
+1. Initialization
+
+    1. Read all input files and store necessary information
+
+    1. Compute mass of each material point
+        $$ m_p = \gamma V_p $$
+
+1. Solution phase for time step $t$ to $t + \Delta t$
+
+    1. Compute nodal mass 
+        $$ m_i^{t+\Delta t} = \Sigma_p N_i(\textbf{x}_p^t) m_p $$
+
+    1. Compute nodal momentum
+        $$ m\textbf{v}_i^{t+\Delta t} = \Sigma_p N_i(\textbf{x}_p^t) m\textbf{v}_p $$
```

Comment:
Missing `m_i` index i for node. Distinguish node mass from particle mass.

Reply:
I want to do "momentum" which is mv, thus I dont have the subscript i